proto/substrait/type.proto

```diff
@@ -159,6 +161,12 @@ message Type {
     Nullability nullability = 4;
   }
 
+  message ParametrizedTimestamp {
```

Comment:
Is this a duplicate of the one in parameterized_types.proto?

Reply:
Since parameterized_types.proto already uses Parameterized as a prefix in all the names, it'll be less confusing to use something like PrecisionTimestamp instead